nat/nat.go

```diff
@@ -61,75 +55,72 @@ func NewPortMapper(log logging.Logger, r Router) Mapper {
 	}
 }
 
-// Map sets up port mapping using given protocol, internal and external ports
-// and returns the final port mapped. It returns 0 if mapping failed after the
-// maximun number of retries
-func (dev *Mapper) Map(protocol string, intPort uint16, desc string) (uint16, error) {
-	mappedPort := make(chan uint16)
+// Attempt to establish a PnP connection from extPort (exposed to the internet) to our
+// intPort (where our process is listening).
+func (dev *Mapper) Map(protocol string, intPort, extPort uint16, desc string) {
+	if !dev.r.IsPnP() {
+		return
```

Comment:
Can we add a log that says "router does not exist or does support PnP. Not performing port mapping" or similar?

Reply:
This actually is b/c of the interface of the router.  And there is a 'NoRouter'...  Which is why I did the IsPnP() check here.  If PnP is setup then you will get an error if it didn't work.  (in the lines you noted below)..

Reply:
I'm not sure I understand...what I mean is, if someone boots their node without specifying `--public-ip` and their router doesn't support PnP. As the code is, this would result in nothing being printed to the user indicating that the port mapping has not been performed, right?

Reply:
you would actually see: invalid IP Address '...' from params.go.
We don't percolate consensusIP (which is the --public-ip flag) into main module.
it's a bit wonky but you would know if PnP didn't work.  I will add more logging into params.go if we don't get a hold of a router..

Reply:
changed this to log an error when PnP failed if it was tried.